src/components/BlockSelector/BlockSelector.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <ToggleInput
+    :id="id"
+    class="block-selector"
+    :class="[
+      {
+        'block-selector--selected': checked,
+        'block-selector--focus': isFocused,
+        'block-selector--disabled': disabled,
+      },
+      `block-selector--${labelPosition}`,
+    ]"
+    :type="type"
+    :data-checked="`${checked}`"
+    :checked="checked"
+    :disabled="disabled"
+    :label-position="labelPosition"
+    :name="name"
+    @focus="isFocused = true"
+    @blur="isFocused = false"
+    @input="$emit('input', $event.target.checked)"
+  >
+    <div class="block-selector__text">
+      <div v-if="!!$slots.label || label">
+        <TypeBody
+          tag="div"
+          variant="medium"
+          weight="strong"
+          class="block-selector__text__label"
+        >
+          <slot name="label">
+            {{ label }}
+          </slot>
+        </TypeBody>
+      </div>
+      <div v-if="!!$slots.description || description">
+        <TypeBody
+          tag="div"
+          variant="text"
+          weight="regular"
+          class="block-selector__text__description"
+        >
+          <slot name="description">
+            {{ description }}
+          </slot>
+        </TypeBody>
+      </div>
```

Comment:
Similarly, this can be simplified to

```suggestion
      <div v-if="!!$slots.description || description" class="block-selector__text__description">
        <slot name="description">
          {{ description }}
        </slot>
      </div>
```

with the style

```
.block-selector__text__description {
  @apply font-body-text text-gray-2 mt-1;
}
```